src/core.ts

```diff
@@ -96,7 +116,22 @@ export async function crawl(config: Config) {
             exclude:
               typeof config.exclude === "string"
                 ? [config.exclude]
-                : config.exclude ?? [],
+                : config.exclude,
+            transformRequestFunction: (req) => {
+              // Additional check for excluded query parameters
+              if (config.exclude && Array.isArray(config.exclude)) {
+                const url = new URL(req.url);
+                for (const pattern of config.exclude) {
+                  if (typeof pattern === "string" && pattern.includes("&do=")) {
```

Comment:
I suppose `&do=` is specific to some test case? Shouldn't the `exclude` config option take care of excluding any pattern?